### DISCUSSION TITLE
Opengraph metadata custom property fields

### DISCUSSION THREAD

**Discussion by mrsafalpiya** [vercel/next.js#55043] [App Router] [answered]
I'm using the Next.js 13 with app directory and I want to add opengraph metadata.

Adding title, description and url is straight forward using the following code:

```typescript
export async function generateMetadata(props: Props): Promise<Metadata> {
  return {
    openGraph: {
      title: "foo",
      description: "bar",
      url: "buzz",
    },
  }
}
```

which creates the following output:

```html
<meta property="og:title" content="foo">
<meta property="og:description" content="bar">
<meta property="og:url" content="buzz">
```

But is there any way to create metatags with the property of `product:brand`, `product:availability` and so on?

The expected output is on the attached image below.

![image](https://github.com/vercel/next.js/assets/66555275/fb9af771-9701-4988-b7e0-edb3072a0b35)

## Things I have tried so far

Using the `other` field as

```typescript
export const metadata = {
  other: {
    custom: 'meta',
  },
}
```

It generates the following output:

```html
<meta name="custom" content="meta" />
```

But I want the output as

```html
<meta property="custom" content="meta" />
```

**Comment by mrsafalpiya** [comment, marked as answer]
The solution was to simple add the `<meta property="foo" content="bar" />` in the page itself.

Example:

```typescript
export default function Page() {
    return (
        <>
            <meta property="foo" content="bar" />
            
            {/* actual page contents */}
        </>
    );
}
```

Thanks to `joulev` in discord who found the fix at [this help-forum](https://discord.com/channels/752553802359505017/1148886601682927646).

Note that `joulev` quoted:

```
note that it is not documented so it may break at any moment, add a test case to check for this in your code
because who knows, next nextjs canary might break it
```

**Reply by dsimmons** [reply]
Thanks so much for this!

I just wasted a lot of time trying all of the same things (also the old `<Head>` tag) before I ultimately found this answer/solution! The fact that custom `openGraph: {}` fields are silently ignored is very unintuitive to say the least!

It'd be great for this to be documented or, even better, officially supported via the `metadata` export (including types)!

**Reply by dsimmons** [reply]
For posterity, I've created a follow-up Discussion thread!

https://github.com/vercel/next.js/discussions/57941
